src/Processors/Formats/Impl/TabSeparatedRowInputFormat.cpp

```diff
@@ -38,7 +38,7 @@ static void checkForCarriageReturn(ReadBuffer & in)
         throw Exception("\nYou have carriage return (\\r, 0x0D, ASCII 13) at end of first row."
             "\nIt's like your input data has DOS/Windows style line separators, that are illegal in TabSeparated format."
             " You must transform your file to Unix format."
-            "\nBut if you really need carriage return at end of string value of last column, you need to escape it as \\r.",
+            "\nBut if you really need carriage return at end of string value of last column, you need to escape it as \\\\r.",
```

Comment:
This change looks incorrect.
The user can write

Hello, world\r
Hello
World

If they indend to add carriage return inside the first field.

Reply:
If the user writes their values in string literals in C, PHP, Perl, JavaScript,
using `\\r` as an example is reasonable.

But this error message is about data contents.

Reply:
But escaping of `\r` is done with double backslash i.e. `\\r`.
Otherwise this is "escape" (as stated in the message).

And besides this `\\r` will be interpreted as `\r` by `parseComplexEscapeSequence` -> `parseEscapeSequence`
```
$ echo -n $'key\\r\nbar\r\n' | clickhouse-local --input-format TSV -S 'key String' -q 'select * from table limit 10 format RowBinary' | xxd
00000000: 046b 6579 0d04 6261 720d                 .key..bar.
                    ^^          ^^
```

Reply:
>But this error message is about data contents.

Ok